pyvo/dal/sia2.py

```diff
@@ -0,0 +1,1064 @@
+    :py:mod:`~astropy.io.votable`
+    :py:class:`~astropy.io.votable.tree.Table` instance where the
+    columns contain the various metadata describing the images.
+    One can access that VOTable directly via the
+    :py:attr:`~pyvo.dal.query.DALResults.votable` attribute.  Thus,
+    when one retrieves a whole column via
+    :py:meth:`~pyvo.dal.query.DALResults.getcolumn`, the result is
+    a Numpy array.  Alternatively, one can manipulate the results
+    as an Astropy :py:class:`~astropy.table.table.Table` via the
+    following conversion:
+
+    >>> table = results.votable.to_table()
+
+    ``SIAResults`` supports the array item operator ``[...]`` in a
+    read-only context.  When the argument is numerical, the result
+    is an
+    :py:class:`~pyvo.dal.sia2.ObsCoreRecord` instance, representing the
+    record at the position given by the numerical index.  If the
+    argument is a string, it is interpreted as the name of a column,
+    and the data from the column matching that name is returned as
+    a Numpy array.
+    """
+
+    def getrecord(self, index):
+        """
+        return a representation of a sia result record that follows
+        dictionary semantics. The keys of the dictionary are those returned by
+        this instance's fieldnames attribute. The returned record has
+        additional image-specific properties
+
+        Parameters
+        ----------
+        index : int
+           the integer index of the desired record where 0 returns the first
+           record
+
+        Returns
+        -------
+        ObsCoreMetadataRecord
+           a dictionary-like wrapper containing the result record metadata.
+
+        Raises
+        ------
+        IndexError
+           if index is negative or equal or larger than the number of rows in
+           the result table.
+
+        See Also
+        --------
+        Record
+        """
+        return ObsCoreRecord(self, index, session=self._session)
+
+
+class ObsCoreRecord(SodaRecordMixin, DatalinkRecordMixin, Record,
+                            ObsCoreMetadata):
+    """
+    a dictionary-like container for data in a record from the results of an
+    image (SIAv2) search, describing an available image in ObsCore format.
+
+    The commonly accessed metadata which are stadardized by the SIA
+    protocol are available as attributes.  If the metadatum accessible
+    via an attribute is not available, the value of that attribute
+    will be None.  All metadata, including non-standard metadata, are also
+    acessible via the ``get(`` *key* ``)`` function (or the [*key*]
+    operator) where *key* is table column name.
+    """
+
+    ###          OBSERVATION INFO
+    @property
+    def data_type(self):
+        """
+        Data product (file content) primary type. This is coded as a string
+        that conveys a general idea of the content and organization of a
+        dataset.
+        """
+        return self['dataproduct_type'].decode('utf-8')
```

Comment:
the `get` method of `Record` has a parameter `decode` which already takes care of decoding.

Reply:
Oh. That's handy. Thanks